code/controllers/subsystem/cleanup.dm

```diff
@@ -0,0 +1,44 @@
+/**
+ * # Cleanup Subsystem
+ *
+ * For now, all it does is periodically clean the supplied global lists of any null values they may contain.
+ *
+ * Why is this important?
+ *
+ * Sometimes, these lists can gain nulls due to errors.
+ * For example, when a dead player trasitions from the `dead_mob_list` to the `alive_mob_list`, a null value may get stuck in the dead mob list.
+ * This can cause issues when other code tries to do things with the values in the list, but are instead met with null values.
+ * These problems are incredibly hard to track down and fix, so this subsystem is a solution to that.
+ */
+SUBSYSTEM_DEF(cleanup)
+	name = "Null cleanup"
+	wait = 30 SECONDS
+	flags = SS_POST_FIRE_TIMING
+	priority = FIRE_PRIORITY_CLEANUP
+	init_order = INIT_ORDER_CLEANUP
+	runlevels = RUNLEVEL_LOBBY | RUNLEVELS_DEFAULT
+	offline_implications = "Certain global lists will no longer be cleared of nulls, which may result in runtimes. No immediate action is needed."
+	/// A list of global lists we want the subsystem to clean.
+	var/list/lists_to_clean
+
+/datum/controller/subsystem/cleanup/Initialize(start_timeofday)
+	. = ..()
```

Comment:
I know its essentially the same, but its standard format here to have `return ..()` at the end of SS init. It also makes more sense to send the initialization message *after* the SS has initialized, not before the proc finishes.

Reply:
Ah good to know